### ISSUE TITLE
issue with authorizing credentials 

### ISSUE THREAD

**Issue by bradboldenow** [burnash/gspread#718]
I am running the basic authentication process (after generating the OAuth credentials outlined in the documentation: https://gspread.readthedocs.io/en/latest/oauth2.html

Here is my code for reference:
```
import gspread
from oauth2client.service_account import ServiceAccountCredentials

scope = ['https://spreadsheets.google.com/feeds',
         'https://www.googleapis.com/auth/drive']

credentials = ServiceAccountCredentials.from_json_keyfile_name('my_creds-06044a753234.json', scope)

gc = gspread.authorize(credentials)

wks = gc.open("My Sheet").sheet1
```

I am receiving a type error, saying the argument for the authorize function in my code is a string. Has anyone seen this issue? I was not able to find any other issues where this was replicated. I am using python 3.7 and am running this code on python anywhere: https://www.pythonanywhere.com/ Any ideas would be greatly appreciated. 

Full Error:
```
Traceback (most recent call last):
  File "/home/data/DTR_GoogleSheets/test.py", line 9, in <module>
    gc = gspread.authorize(credentials)
  File "/home/data/.local/lib/python3.7/site-packages/gspread/__init__.py", line 38, in authorize
    client.login()
  File "/home/data/.local/lib/python3.7/site-packages/gspread/client.py", line 51, in login
    self.auth.refresh(http)
  File "/home/data/.local/lib/python3.7/site-packages/oauth2client/client.py", line 545, in refresh
    self._refresh(http)
  File "/home/data/.local/lib/python3.7/site-packages/oauth2client/client.py", line 749, in _refresh
    self._do_refresh_request(http)
  File "/home/data/.local/lib/python3.7/site-packages/oauth2client/client.py", line 780, in _do_refresh_request
    body=body, headers=headers)
  File "/home/data/.local/lib/python3.7/site-packages/oauth2client/transport.py", line 282, in request
    connection_type=connection_type)
  File "/usr/lib/python3.7/site-packages/httplib2/__init__.py", line 1957, in request
    cachekey,
  File "/usr/lib/python3.7/site-packages/httplib2/__init__.py", line 1622, in _request
    conn, request_uri, method, body, headers
  File "/usr/lib/python3.7/site-packages/httplib2/__init__.py", line 1528, in _conn_request
    conn.connect()
  File "/usr/lib/python3.7/site-packages/httplib2/__init__.py", line 1309, in connect
    sock.connect((self.host, self.port))
  File "/usr/lib/python3.7/site-packages/socks.py", line 406, in connect
    self.__negotiatehttp(destpair[0],destpair[1])
  File "/usr/lib/python3.7/site-packages/socks.py", line 357, in __negotiatehttp
    while resp.find("\r\n\r\n")==-1:
TypeError: argument should be integer or bytes-like object, not 'str'
```

**Comment by CubeBlazer** [comment]
Try opening your sheet by key using `sh = gc.open_by_key(INSERT YOUR KEY HERE)` then `wks = sh.worksheet(INSERT WORKSHEET NAME)

**Comment by bradboldenow** [comment]
Hi @CubeBlazer, thanks for the response. The issue doesn't seem to be with opening the sheet. It seems to be in authorizing using my credentials ( i.e. gc = gspread.authorize(credentials) is the failure point based on the error.)

**Comment by msvanberg** [comment]
FWIW having the same problem, also on PythonAnywhere. It runs on my local machine though...

**Comment by danspanner** [comment]
I am also having auth issues.

`#!/usr/bin/env python3
  
import gspread
from oauth2client.service_account import ServiceAccountCredentials
scope = ['https://spreadsheets.google.com/feeds','https://googleapis.com/auth/drive']
#scope = ['https://googleapis.com/auth/spreadsheets']
credentials = ServiceAccountCredentials.from_json_keyfile_name('service.json', scope)
gc = gspread.authorize(credentials)
#wks = gc.open("Test API Sheet").sheet1
sht1 = gc.open_by_key("1i9n3GRMa33H8...9yFiBxaXU")
`

After following the Read the Docs page here
https://gspread.readthedocs.io/en/latest/oauth2.html

`File "./test.py", line 12, in <module>
    gc = gspread.authorize(credentials)
  File "/home/dan/Documents/code/gsheets_poc/lib/python3.7/site-packages/gspread/__init__.py", line 38, in authorize
    client.login()
  File "/home/dan/Documents/code/gsheets_poc/lib/python3.7/site-packages/gspread/client.py", line 51, in login
    self.auth.refresh(http)
  File "/home/dan/Documents/code/gsheets_poc/lib/python3.7/site-packages/oauth2client/client.py", line 545, in refresh
    self._refresh(http)
  File "/home/dan/Documents/code/gsheets_poc/lib/python3.7/site-packages/oauth2client/client.py", line 749, in _refresh
    self._do_refresh_request(http)
  File "/home/dan/Documents/code/gsheets_poc/lib/python3.7/site-packages/oauth2client/client.py", line 819, in _do_refresh_request
    raise HttpAccessTokenRefreshError(error_msg, status=resp.status)
oauth2client.client.HttpAccessTokenRefreshError: invalid_scope: Invalid oauth scope or ID token audience provided.
`

I noticed in the documentation that it refers to a "Service Key" as opposed to a Service Account- is it possible the documentation is out of date and there are further steps to the auth process?

**Comment by msvanberg** [comment]
My issue was fixed by running `pip3.7 install --user --force-reinstall PySocks` in a Bash console.

**Comment by alihassan10020161** [comment]
check this link. it resolve my same problem 

https://www.youtube.com/watch?v=cnPlKLEGR7E

**Comment by kyluca** [comment]
Hey all, I was having similar OAuth issues when trying to update some old code (using gspread==0.6.2) to use the latest versions (gspread>=3.0.0) and I believe it might be the same solution required here.
The scopes OP is using were the same as mine, `https://spreadsheets.google.com/feeds`, which worked for old versions of the Google Sheets API. But now we should be using v4 of the Google Sheets API with the correct scope `https://www.googleapis.com/auth/spreadsheets`.

As per the Google Sheets API docs: https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/get#authorization-scopes
> Requires one of the following OAuth scopes:
> 
> * https://www.googleapis.com/auth/drive
> * https://www.googleapis.com/auth/drive.readonly
> * https://www.googleapis.com/auth/drive.file
> * https://www.googleapis.com/auth/spreadsheets
> * https://www.googleapis.com/auth/spreadsheets.readonly

Hope that helps someone else too.

**TL;DR**: Change `https://spreadsheets.google.com/feeds` to `https://www.googleapis.com/auth/spreadsheets`

**Comment by marius-ne** [comment]
> 
> 
> My issue was fixed by running `pip3.7 install --user --force-reinstall PySocks` in a Bash console on the platform.

This worked for me as well! Had the same issue as OP while running on PythonAnywhere.

**Comment by lavigne958** [comment]
Issues seems resolved. we now use the new scopes for the API V4 and it's all provided by default by gspread suing `gspread.oauth()`.

closing this issue.
